### PR TITLE
[PDI-18080] Premature termination of Executor tabs causes Spoon crash

### PR DESCRIPTION
**Pull request by ppatricio**

### CODE REVIEW

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 18m 25s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl ui
```
---
:ok_hand: All tests passed!
 
Tests run: **1144**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](http://wingman.pentaho.net/job/projects/job/pdi/job/PR-7505/1/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 19m 16s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl ui
```
---
:ok_hand: All tests passed!
 
Tests run: **1144**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](http://wingman.pentaho.net/job/projects/job/pdi/job/PR-7505/2/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

File: ui/src/main/java/org/pentaho/di/ui/job/entries/trans/JobEntryBaseDialog.java [hunk 1]
```diff
@@ -707,14 +707,16 @@ public void modifyText( ModifyEvent modifyEvent ) {
...
         }
-        display.asyncExec( () -> {
-          if ( (Boolean) items.get( IS_PENTAHO ) ) {
-            wWaitingToFinish.setSelection( false );
-            wWaitingToFinish.setEnabled( false );
-          } else {
-            wWaitingToFinish.setEnabled( true );
-          }
-        } );
+        if ( wWaitingToFinish != null && !wWaitingToFinish.isDisposed() ) {
```
Comment (ssamora) [inline ui/src/main/java/org/pentaho/di/ui/job/entries/trans/JobEntryBaseDialog.java:710]: I think this check should be done inside the asyncExec. When we do an asyncExec it does not immediately execute. We are creating a new Runnable that will be run by the UI thread. 
So, in the point where we check the wWaitingToFinish everything might be ok and we create the new async execution. When the async execution actually runs then wWaitingToFinish might already be null or disposed.

Comment (ppatricio) [reply ui/src/main/java/org/pentaho/di/ui/job/entries/trans/JobEntryBaseDialog.java:710]: I understand that, and i also considered that option, but yhe question that came into my mind was, why even create a running if the object that is going to affected is already disposed?

Comment (ppatricio) [reply ui/src/main/java/org/pentaho/di/ui/job/entries/trans/JobEntryBaseDialog.java]: New commit added. Please consider squashing it when merging.

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 16m 50s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl ui
```
---
:ok_hand: All tests passed!
 
Tests run: **1144**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](http://wingman.pentaho.net/job/projects/job/pdi/job/PR-7505/3/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 14m 41s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl ui
```
---
:ok_hand: All tests passed!
 
Tests run: **1144**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](http://wingman.pentaho.net/job/projects/job/pdi/job/PR-7505/4/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 18m 34s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl ui
```
---
:ok_hand: All tests passed!
 
Tests run: **1144**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](http://wingman.pentaho.net/job/projects/job/pdi/job/PR-7505/5/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Pull request merged**